### PR TITLE
Adding comments for EventContext and main CloudFunctions methods.

### PR DESCRIPTION
**Pull request by egilmorez**
Adding TypeDoc-friendly comments for EventContext and main CloudFunctions methods.

### CODE REVIEW

File: src/cloud-functions.ts [hunk 1]
```diff
@@ -179,7 +235,7 @@ export interface Resource {
   labels?: { [tag: string]: string };
 }
 
-/**
+/** @hidden
```
Comment (thechenky) [inline src/cloud-functions.ts]: This should always be on the second line with first line being `/**`, unless it's a one-line comment like `/** @hidden */`

File: src/cloud-functions.ts [hunk 2]
```diff
@@ -53,38 +55,92 @@ export interface Event {
...
+       `projects/_/instances/<databaseInstance>/refs/<databasePath>`
+  * * Storage &mdash;
+      `projects/_/buckets/<bucketName>/objects/<fileName>#<generation>`
+  * * Authentication &mdash; `projects/<projectId>`
+  * * Pub/Sub &mdash; `projects/<projectId>/topics/<topicName>`
+  *
+  * Because Realtime Database instances and Cloud Storage buckets are globally
+  * unique and not tied to the project, their resources start with `projects/_`.
+  * Underscore is not a valid project name.
+  *
```
Comment (thechenky) [inline src/cloud-functions.ts]: Remove extra line.

File: src/cloud-functions.ts [hunk 3]
```diff
@@ -53,38 +55,92 @@ export interface Event {
...
-   * Type of authentication for the triggering action. Only available for
-   * database functions.
+   * The level of permissions for a user. Valid values are:
+   *
+   * * `ADMIN` Developer user or user authenticated via a service account.
+   * * `USER` Known user.
+   * * `UNAUTHENTICATED` Unauthenticated action
+   * * `null` For event types that do not provide user information (all except
+   *   Realtime Database).
+   *
```
Comment (thechenky) [inline src/cloud-functions.ts]: Remove extra line.

File: src/cloud-functions.ts [hunk 4]
```diff
@@ -53,38 +55,92 @@ export interface Event {
...
+  * unique and not tied to the project, their resources start with `projects/_`.
+  * Underscore is not a valid project name.
+  *
    */
   resource: Resource;
   /**
-   * Timestamp for when the event ocurred (ISO 8601 string)
+   * Timestamp for the event as an
+   * [RFC 3339](https://www.ietf.org/rfc/rfc3339.txt) string.
+   *
```
Comment (thechenky) [inline src/cloud-functions.ts]: Not sure about whether this one changes how the docs look but if it doesn't, let's remove this extra line.

File: src/cloud-functions.ts (hunk 1 above)
Comment (egilmorez) [reply src/cloud-functions.ts]: SG.  With no empty lines before nor after?

File: src/cloud-functions.ts (hunk 2 above)
Comment (egilmorez) [reply src/cloud-functions.ts]: Good catch; this was lined up strangely too.

**Pull request merged**